runners/inline.js

```diff
@@ -6,25 +6,29 @@ var readable = require('@momsfriendlydevco/readable');
 module.exports = {
 	id: 'inline',
 	exec: session => new Promise((resolve, reject) => {
-		async.run(session.context, session.worker.worker, (err, value) => {
-			if (err) {
-				reject(err)
-			} else if (_.isString(session.worker.expires) && session.worker.expires) { // Stash value with an expiry
-				var expiry = new Date(Date.now() + (timestring(session.worker.expires) * 1000));
-				session.context.log(`Stashing result with expiry of ${expiry}`);
-				var stashStart = new Date();
-				session.cacher.set(session.cacheKey, value, expiry, err => {
-					session.context.log(`Stash write complete in ${readable.relativeTime(stashStart, {formatters: {fallback: '0ms'}})}`);
-					resolve(value);
-				});
-			} else { // Stash value with no expiry
-				session.context.log('Stashing result with no expiry');
-				var stashStart = new Date();
-				session.cacher.set(session.cacheKey, value, err => {
-					session.context.log(`Stash write complete in ${readable.relativeTime(stashStart, {formatters: {fallback: '0ms'}})}`);
-					resolve(value);
-				});
-			}
-		}, [session.agentSettings, session]);
+		try {
+			async.run(session.context, session.worker.worker, (err, value) => {
+				if (err) {
+					reject(err);
+				} else if (_.isString(session.worker.expires) && session.worker.expires) { // Stash value with an expiry
+					var expiry = new Date(Date.now() + (timestring(session.worker.expires) * 1000));
+					session.context.log(`Stashing result with expiry of ${expiry}`);
+					var stashStart = new Date();
+					session.cacher.set(session.cacheKey, value, expiry, err => {
+						session.context.log(`Stash write complete in ${readable.relativeTime(stashStart, {formatters: {fallback: '0ms'}})}`);
+						resolve(value);
+					});
+				} else { // Stash value with no expiry
+					session.context.log('Stashing result with no expiry');
+					var stashStart = new Date();
+					session.cacher.set(session.cacheKey, value, err => {
+						session.context.log(`Stash write complete in ${readable.relativeTime(stashStart, {formatters: {fallback: '0ms'}})}`);
+						resolve(value);
+					});
+				}
+			}, [session.agentSettings, session]);
+		} catch(err) {
+			reject(err.message);
```

Comment:
Wrapping `inline` in a `try/catch` means it outputs a string like `finish('string', [])` does.